src/interceptor-service.ts

```diff
@@ -1,258 +1,517 @@
 import {
-	ConnectionBackend,
-	Headers,
-	Http,
-	Request,
-	Response,
-	ResponseOptions,
-	RequestMethod,
-	RequestOptions
+  ConnectionBackend,
+  Headers,
+  Http,
+  Request,
+  Response,
+  ResponseOptions,
+  RequestMethod,
+  RequestOptions
 } from '@angular/http';
+import { RequestOptionsArgs } from '@angular/http/src/interfaces';
+
 import { Observable, Observer } from 'rxjs/Rx';
 import { ErrorObservable } from 'rxjs/observable/ErrorObservable';
-import { InterceptedRequest } from "./intercepted-request";
-import { InterceptedResponse } from "./intercepted-response";
-import { InterceptorOptions } from "./interceptor-options";
-import { Interceptor } from "./interceptor";
 
+import { InterceptorRequest } from './interceptor-request';
+import { InterceptorResponseWrapper } from './interceptor-response-wrapper';
+import { Interceptor } from './interceptor';
+import { InterceptorRequestOptionsArgs } from './interceptor-request-options-args';
+import { InterceptorRequestBuilder } from './interceptor-request-builder';
+import { InterceptorResponseWrapperBuilder } from './interceptor-response-wrapper-builder';
+import { InterceptorUtils } from './interceptor-utils';
+import { RealResponseObservableTransformer } from './real-response-observable-transformer';
+
+import './object-assign-polyfill';
```

Comment:
This has to be removed from the code. Dint see any errors by tsc. Forgot to remove